File: app/src/main/java/com/pixels/blockies/app/game/figures/Picker.java
package com.pixels.blockies.app.game.figures;

import java.util.LinkedList;
import java.util.Queue;
import java.util.Random;

public class Picker {
    private Random random = new Random();
    private Queue<Rotatable> queue = new LinkedList<Rotatable>();

    public Picker() {
        queue.add(get());
    }

    private Rotatable get() {
        int num = random.nextInt(6);
        Rotatable picked = null;
        switch (num) {
            case 0:
                picked = new FigureI();
                break;
            case 1:
                picked = new FigureJ();
                break;
            case 2:
                picked = new FigureL();
                break;
            case 3:
                picked = new FigureT();
                break;
            case 4:
                picked = new FigureZ();
                break;
            case 5:
                picked = new FigureO();
                break;
        }
        return picked;
    }

    public synchronized Rotatable pick() {
        queue.add(get());
        return queue.poll();
    }

    public synchronized Rotatable peek() {
        return queue.peek();
    }

}

File: app/src/main/java/com/pixels/blockies/game/game/figures/Picker.kt
package com.pixels.blockies.game.game.figures

import java.util.*

class Picker {
    private val random = Random()
    private val queue: Queue<Rotatable?> = LinkedList()

    init {
        queue.add(get())
    }

    private fun get(): Rotatable? {
        val num = random.nextInt(9)
        var picked: Rotatable? = null
        when (num) {
            0 -> picked = FigureI()
            1 -> picked = FigureJ()
            2 -> picked = FigureL()
            3 -> picked = FigureT()
            4 -> picked = FigureZ()
            5 -> picked = FigureZRevert()
            6 -> picked = FigureO()
            7 -> picked = FigureSplitLine()
            8 -> picked = FigureSingleDot()
        }
        return picked
    }

    @Synchronized
    fun pick(): Rotatable? {
        queue.add(get())
        return queue.poll()
    }

    @Synchronized
    fun peek(): Rotatable? {
        return queue.peek()
    }
}